### PR TITLE
random projection of triaxial ellipsoid

### PR DESCRIPTION
**Pull request by ntessore**
## Description

Factors out the random projection of the ellipsoid from `skypy.galaxy.ellipticity.ryden04` into a random sampler `skypy.utils.random.triaxial_axis_ratio`.

This might be reused for other ellipticity sampling for galaxies (@jucordero) or halos and clusters (@sutieng).

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/utils/random.py [hunk 1]
```diff
@@ -73,3 +74,56 @@ def schechter(alpha, x_min, x_max, resolution=100, size=None, scale=1.):
...
+    Parameters
+    ----------
+    zeta : array_like
+        Axis ratio of intermediate and major axis.
+    xi : array_like
+        Axis ratio of minor and major axis.
+    size : tuple of int or None
+        Size of the random draw. If `None` is given, size is inferred from
+        other inputs.
+
```
Comment (Lucia-Fonseca) [inline skypy/utils/random.py:94]: is `Returns` section missing?

File: skypy/utils/tests/test_random.py [hunk 2]
```diff
@@ -56,3 +56,38 @@ def test_schechter_gamma():
 
     _, p = kstest(x, 'gamma', args=(alpha+1, 0, scale))
     assert p > 0.01
+
+
+@pytest.mark.flaky
+def test_triaxial_axis_ratio():
+
+    from skypy.utils.random import triaxial_axis_ratio
```
Comment (Lucia-Fonseca) [inline skypy/utils/tests/test_random.py:64]: Should we do all the imports at the top of file?

File: skypy/utils/random.py (hunk 1 above)
Comment (ntessore) [reply skypy/utils/random.py:94]: Yes, let me try and add one
```suggestion

    Returns
    -------
    q : array_like
        Axis ratio of the randomly projected ellipsoid.

```

File: skypy/utils/tests/test_random.py [hunk 3]
```diff
@@ -56,3 +56,38 @@ def test_schechter_gamma():
 
     _, p = kstest(x, 'gamma', args=(alpha+1, 0, scale))
     assert p > 0.01
+
+
+@pytest.mark.flaky
```
Comment (rrjbca) [inline skypy/utils/tests/test_random.py:61]:
```suggestion
```

File: skypy/galaxy/ellipticity.py [hunk 4]
```diff
@@ -120,19 +121,8 @@ def ryden04(mu_gamma, sigma_gamma, mu, sigma, size=None):
...
-
-    # compute Binney's ABC -- eq.s (13)-(15)
-    A = (1 - eps*(2-eps)*sin2_phi)*cos2_theta + gam**2*sin2_theta
-    B = (2*eps*(2-eps))**2*cos2_theta*sin2_phi*cos2_phi
-    C = 1 - eps*(2-eps)*cos2_phi
-
-    # compute axis ratio q -- eq. (12)
-    q = np.sqrt((A+C-np.sqrt((A-C)**2+B))/(A+C+np.sqrt((A-C)**2+B)))
+    # random projection of random triaxial ellipsoid
+    q = random.triaxial_axis_ratio(gam, 1-eps)
```
Comment (ntessore) [inline skypy/galaxy/ellipticity.py:125]: Wrong order: `gam = c/a`, `eps = 1 - b/a` for `a >= b >= c` (Ryden 2004).
```suggestion
    q = random.triaxial_axis_ratio(1-eps, gam)
```

**Review by rrjbca — APPROVED** [review]

**Pull request merged**